src/components/FruitBasket.js

```diff
@@ -0,0 +1,86 @@
+import React, {useState}  from 'react';
+import './FruitBasket.css';
+import Fruit from './Fruit.js';
+
+
+
+
+function FruitBasket() {
+
+    const [counterApple, setCounterApple] = useState(0);
+    const [counterKiwi, setCounterKiwi] = useState(0);
+    const [counterStrawberry, setcounterStrawberry] = useState(0);
+    const [counterBanana, setcounterBanana] = useState(0);
+
+
+    function add({id}) {
+
+        switch (id) {
+            case "Appels":
+                setCounterApple(counterApple + 1);
+                break;
+            case "Aarbeien":
+                setcounterStrawberry(counterStrawberry + 1);
+                break;
+            case "Bananen":
+                setcounterBanana(counterBanana + 1);
+                break;
+
+            default:
+                setCounterKiwi(counterKiwi + 1);
+        }
+    }
+
+    function subtract({id, counter}) {
+
+        switch (counter) {
+            case 0:
+                // counter cannot be lower then zero
+                break;
+
+            default:
+
+                switch (id) {
+                    case "Appels":
+                        setCounterApple(counterApple - 1);
+                        break;
+                    case "Aarbeien":
+                        setcounterStrawberry(counterStrawberry - 1);
+                        break;
+                    case "Bananen":
+                        setcounterBanana(counterBanana - 1);
+                        break;
+
+                    default:
+                        setCounterKiwi(counterKiwi - 1);
+                }
+        }
+    }
```

Comment:
Ik waardeer het idee met die `id`, maar ik had zelf gewoon de counter en set-functie meegegeven aan het component. En dan in het fruit component zelf checken of de counter > 0 is enzo. Anders gezegd: Je zou nu per fruitsoort die je toevoegd 7 regels code erbij krijgen terwijl als je de counter en functie meegeeft aan het component je maar één regel extra zou krijgen